src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6872,7 +6884,9 @@ private void VisitLambda(BoundLambda node, NamedTypeSymbol delegateTypeOpt, Diag
                 initialState: initialState ?? GetVariableState(State.Clone()),
                 analyzedNullabilityMap,
                 snapshotBuilder,
-                returnTypesOpt: null);
+                returnTypesOpt: null,
+                localFuncVarUsagesOpt: _localFuncVarUsages ??= new SmallDictionary<LocalFunctionSymbol, LocalFunctionState>()
```

Comment:
Why do we need an explicit dictionary here instead of just using `null`?

Reply:
Here if the parent passes a null dictionary to the child, the child will simply initialize its own `localFuncVarUsages` on-demand and throw it away once finished.

As an alternative to this, if the parent has a null `localFuncVarUsages` it could try to extract a non-null `localFuncVarUsages` from the child after lambda analysis is done, or it could just avoid sharing an instance with the child and merge the child's entries into the parent's instance once it is done.

Reply:
This is a mutable type. Are there dangers sharing this with child walkers?